src/client/api/object.c

```diff
@@ -382,33 +426,41 @@ daos_oit_open(daos_handle_t coh, daos_epoch_t epoch,
 	}
 
 	if (co_global_ver < 2) {
-		oid = daos_oit_gen_id(epoch, cont_rf);
-		goto open;
+		rc = dc_obj_open_task_create(coh, daos_oit_gen_id(epoch, cont_rf),
+					     DAOS_OO_RW, oh, ev, NULL, &task);
+		if (rc)
+			return rc;
+
+		return dc_task_schedule(task, true);
+	}
+
+	D_ALLOC_PTR(ooa);
+	if (ooa == NULL) {
+		rc = -DER_NOMEM;
+		goto failed;
 	}
 
-	rc = dc_task_create(dc_cont_snap_oit_oid_get, NULL, NULL, &task);
+	rc = dc_task_create(dc_cont_snap_oit_oid_get, NULL, ev, &task);
```

Comment:
hmm you are using ev twice here for the same  task.. it looks like ev would be completed in the first task though. so not sure how this would work properly. user might poll ev and get completion when first task completes, but clearly it's not complete yet.
so IMO need 1 api_task for ev, then 2 subtasks:
open_task
oit_get_tasks
with dependency between them.
open_task->oit_get_task->api_task
and then can call schedule on task after dependency and cbs are registered

Reply:
as discussed, if co_global_ver < 2, it will return where first ev used.

Reply:
yes sorry. i missed return call there.